pkg/api/controller.go

```diff
@@ -3068,14 +3113,17 @@ func (c *Controller) UpdatePassword(w http.ResponseWriter, r *http.Request, body
 		writeError(w, http.StatusUnauthorized, err)
 		return
 	}
-	tokenEmail := claims.Subject
-	user, err := c.Auth.GetUserByEmail(r.Context(), tokenEmail)
+	user, err := c.Auth.GetUserByEmail(r.Context(), claims.Subject)
 	if err != nil {
-		c.Logger.WithError(err).WithField("email", tokenEmail).Debug("failed to retrieve user by email")
+		c.Logger.WithError(err).WithField("email", claims.Subject).Debug("failed to retrieve user by email")
+		writeError(w, http.StatusNotFound, http.StatusText(http.StatusNotFound))
+		return
+	}
+	if user.Username != normalizeAndSetToLowercase(claims.Subject) {
```

Comment:
Why do we need this one? If the user got a token that allows a change of password - any user that we will get back from `GetUserByEmail` should be matched this description.

Reply:
True, Removed